.golangci.yml

```diff
@@ -37,15 +37,467 @@ linters:
+    - gosimple # Linter for Go source code that specializes in simplifying code
+    - govet # Vet examines Go source code and reports suspicious constructs, such as Printf calls whose arguments do not align with the format string
+    - typecheck # Like the front-end of a Go compiler, parses and type-checks Go code
+    - unused # Checks Go code for unused constants, variables, functions and types
+    - containedctx # containedctx is a linter that detects struct contained context.Context field
+    - cyclop # checks function and package cyclomatic complexity
+    - durationcheck # check for two durations multiplied together
+    - errchkjson
+    - gochecknoglobals # check that no global variables exist
+    - goerr113 # Golang linter to check the errors handling expressions
+    - gomnd # An analyzer to detect magic numbers.
+    - ireturn # Accept Interfaces, Return Concrete Types
+    - nosprintfhostport # Checks for misuse of Sprintf to construct a host with port in a URL.
+    - promlinter # Check Prometheus metrics naming via promlint
+    - reassign # Checks that package variables are not reassigned
+    - revive # Fast, configurable, extensible, flexible, and beautiful linter for Go. Drop-in replacement of golint.
+    - usestdlibvars # A linter that detect the possibility to use variables/constants from the Go standard library.
+
+    # is disabled because of generics. You can track the evolution of the generics support by following the https://github.com/golangci/golangci-lint/issues/2649
+    #- wastedassign # wastedassign finds wasted assignment statements.
 
 linters-settings:
   gofmt:
     simplify: true
+
   goconst:
     min-len: 3
     min-occurrences: 3
+    numbers: true
+
+  goimports:
+    local-prefixes: github.com/0xPolygon/go-ibft
+
+  nestif:
+    min-complexity: 3
+
+  prealloc:
+    for-loops: true
+
+  gocritic:
+    # Which checks should be enabled; can't be combined with 'disabled-checks';
+    # See https://go-critic.github.io/overview#checks-overview
+    # To check which checks are enabled run `GL_DEBUG=gocritic ./build/bin/golangci-lint run`
+    # By default list of stable checks is used.
+    enabled-checks:
+      - badLock
+      - filepathJoin
+      - sortSlice
+      - sprintfQuotedString
+      - syncMapLoadAndDelete
+      - weakCond
+      - boolExprSimplify
+      - httpNoBody
+      - ioutilDeprecated
+      - nestingReduce
+      - preferFilepathJoin
+      - redundantSprint
+      - stringConcatSimplify
+      - timeExprSimplify
+      - typeAssertChain
+      - yodaStyleExpr
+      - truncateCmp
+      - equalFold
+      - preferDecodeRune
+      - preferFprint
+      - preferStringWriter
+      - preferWriteByte
+      - sliceClear
+      #- ruleguard
+
+    # Which checks should be disabled; can't be combined with 'enabled-checks'; default is empty
+    disabled-checks:
```

Comment:
Is this list reflecting current code? Why not leave this empty?

Reply:
the list of disabled checks is from Bor+a few projects. We can discuss each in the list. I strongly encourage to try and check if the linter is useful or not.

Reply:
I'm with Victor: maybe we should consider running a default configuration (keep both `enabled-checks` and `disabled-checks` empty). WDYT?

Reply:
how we'd have things like `unlambda`?

Reply:
Did you try with an empty list? does it reports many errors? This looks good https://go-critic.com/overview.html#unlambda-ref

Reply:
Ok, more linters not less. You, as a team, always can switch some of them off, after a discussion.

Reply:
Using defaults https://go-critic.com/overview.html#checkers and enabling some extra ones we need

Reply:
Because the [default](https://go-critic.com/overview.html#unlambda-ref:~:text=All%20diagnostics%20are%20enabled%20by%20default%20(unless%20it%20has%20%E2%80%9Cexperimental%E2%80%9D%20tag).) ones are enabled if nothing is specified. That's why I was thinking about having both `enabled-checks` and `disabled-checks` empty, because I was thinking it is good enough to run on default rule set. Basically all of non-experiment rules are enabled by default.

Reply:
Experimental != Bad, like `badLock` linter

Reply:
I'd start with as full list as possible and then remove that works not very good for the project. Such things like checking mutexes, I'd rather keep.